libr/util/lib.c

```diff
@@ -303,19 +303,82 @@ R_API int r_lib_open_ptr (RLib *lib, const char *file, void *handler, RLibStruct
 
 	return ret;
 }
+#if __WINDOWS__ && !defined(__CYGWIN__)
+static char *utf16_to_utf8 (const wchar_t *wc) {
+	char *rutf8;
+	int csize;
+	
+	rutf8 = NULL;
+	csize = WideCharToMultiByte (CP_UTF8, 0, wc, -1, NULL, 0, NULL, NULL);
+	if (csize > 0) {
+		rutf8 = malloc (csize);
+		WideCharToMultiByte (CP_UTF8, 0, wc, -1, rutf8, csize, NULL, NULL);
+	}
+	return rutf8;
+}
+
+static wchar_t *utf8_to_utf16 (const char *cstring) {
+	wchar_t *rutf16;
+	int wcsize;
+	
+	rutf16 = NULL;
+	wcsize = MultiByteToWideChar (CP_UTF8, 0, cstring, -1, NULL, 0); 
+	if ( wcsize > 0 ) {
+		rutf16 = (wchar_t *) malloc (sizeof (wchar_t *) * wcsize);
```

Comment:
This will allocate a buffer 2x to big. `MultiByteToWideChar` returns the number of bytes required to fit the output string, not the number of characters.